components/brave_wallet_ui/components/shared/untrusted-image/untrusted-image.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+// utils
+import { httpifyIpfsUrl, stripERC20TokenImageURL } from '../../../utils/string-utils'
+
+// style
+import {
+  LoadingOverlay,
+  UntrustedImageIframe,
+  LoadIcon
+} from './untrusted-image.styles'
+
+interface Props {
+  src?: string
+  responsive?: boolean
+  width?: string | number
+  height?: string | number
+}
+
+export const UntrustedImage = ({
+  src,
+  responsive,
+  height,
+  width
+}: Props) => {
+  // state
+  const [isLoaded, setIsLoaded] = React.useState<boolean>(false)
+
+  const imgURL = stripERC20TokenImageURL(
+    httpifyIpfsUrl(src)
+  )
+
+  return <>
+    <LoadingOverlay isLoading={!isLoaded}>
+      <LoadIcon />
+    </LoadingOverlay>
+
+    <UntrustedImageIframe
```

Comment:
if there is any noticeable performance problem from having 1 iframe per icon in this page, i'd suggest just a single iframe for the UI element that includes all these images.

Reply:
The images take longer to load-in but the iframes are setup for lazy loading so that they should not delay the main UI. If we need iframes for elements containing token lists, we will need multiple of these untrusted iframes setup